sdcm/cluster_azure.py

```diff
@@ -84,8 +87,24 @@ def is_spot(self):
     def check_spot_termination(self):
         """Check if a spot instance termination was initiated by the cloud.
         """
-        # todo: fix it
-        return False
+        try:
+            self.wait_ssh_up(verbose=False)
+            result = self.remoter.run(
+                'curl http://169.254.169.254/metadata/scheduledevents?api-version=2020-07-01 -H "Metadata: true"', verbose=False)
+            status = json.loads(result.stdout.strip())
+            if status["DocumentIncarnation"] == self.scheduled_events_document_incarnation:
+                return SPOT_TERMINATION_CHECK_OVERHEAD
+            for event in status["Events"]:
+                self.scheduled_events_document_incarnation = status["DocumentIncarnation"]
+                if event["EventType"] == "Preempt":
+                    message = f"got spot termination event for node: {event['Resources']}. VM will be evicted at: {event['NotBefore']} "
+                    SpotTerminationEvent(node=self, message=message).publish()
+                else:
```

Comment:
IIUC, we will enter here always we didn't get `event["EventType"] == "Preempt"`
is it the right thing we need to do? meaning, we expect not to get in the check above frequently, but the else will happen every time we check for spot termination, so the following message isn't good, is it?

Reply:
No, when there's no events, content of for loop will be not executed and it won't enter else statement. For each event we test EventType and do proper action.
For now let's raise a warning for other events that Azure can trigger on our machines due maintenance. When we see them, we can adapt code to do proper action then. I added some comment line to this.